VisioShape.ps1

```diff
@@ -130,3 +130,202 @@ Function Get-VisioShape{
+
+        .PARAMETER Master
+        Either the name of the new rack equipment shape master (previously registered using Register-VisioShape) or a reference to a master object.
+
+        .PARAMETER Label
+        The label for the new rack equipment shape.
+
+        .PARAMETER Name
+        The name for the new rack equipment shape. If not provided then the Label will be used as the Name
+
+        .PARAMETER RackLabel
+        The name for the existing rack shape.
+
+        .PARAMETER RackVendor
+        The vendor name for the existing rack shape.
+
+        Supported Rack Vendors are HPE, Dell, IBM & Cisco.
+
+        Within each vendor only the shapes listed below are working.
+
+            HPE shapes from the HPE-Racks set on VisioCafe - https://www.visiocafe.com/downloads/hp/HPE-Common.zip
+
+                    HPE 22U G2 Adv Rack Front
+                    HPE 22U G2 Adv Rack Rear
+                    HPE 36U G2 Adv Rack Front
+                    HPE 36U G2 Adv Rack Rear
+                    HPE 42U 800mm G2 Adv Rack Front
+                    HPE 42U 800mm G2 Adv Rack Rear
+                    HPE 42U 800mm G2 Ent Rack Front
+                    HPE 42U 800mm G2 Ent Rack Rear
+                    HPE 42U G2 Adv Rack Front
+                    HPE 42U G2 Adv Rack Rear
+                    HPE 42U G2 Ent Rack Front
+                    HPE 42U G2 Ent Rack Rear
+                    HPE 48U 800mm G2 Adv Rack Front
+                    HPE 48U 800mm G2 Adv Rack Rear
+                    HPE 48U 800mm G2 Ent Rack Front
+                    HPE 48U 800mm G2 Ent Rack Rear
+                    HPE 48U G2 Adv Rack Front
+                    HPE 48U G2 Adv Rack Rear
+                    HPE 48U G2 Ent Rack Front
+                    HPE 48U G2 Ent Rack Rear
+                    50U Ent. Rack
+
+            Dell shapes from the Dell-Racks set on VisioCafe - https://www.visiocafe.com/downloads/dell/Dell-Racks.zip
+
+                    2420 Rack Frame
+                    4220 Rack Frame
+                    4220D Rack Frame
+                    4220W Rack Frame
+                    4820 Rack Frame
+                    4820D Rack Frame
+                    4820W Rack Frame
+
+            IBM shapes from the IBM-Racks set on VisioCafe - https://www.visiocafe.com/downloads/ibm/IBM-Common.zip
+
+                    7014-S11 Rack
+                    7014-S25 Rack
+                    7014-S00 Rack
+                    7014-T00 Rack
+                    7014-T42 Rack
+
+            Cisco shapes from the Cisco R-Series set on Cisco.com - https://www.cisco.com/c/dam/assets/prod/visio/visio/racks-cisco-r-series.zip
+
+                    RACK2-UCS Front
+                    RACK2-UCS Rear
+                    RACK2-UCS2 Front
+                    RACK2-UCS2 Rear
+
+        .PARAMETER FirstU
+        The first (lowest) U space in which to place the new rack equipment shape.
+
+        .PARAMETER X

```

Comment:
confused by the x and y parameters.  You're going to move the shape automatically, so why not just use these default values and not have the user supply them?

Reply:
I decided it would be nice to allow the user to chose where the initial shape is dropped prior to connecting it to the rack shape. Parameters `$X` and `$Y` are not mandatory. If they are not supplied they default to `$X=1.00` and `$Y=0.25`.

Reply:
If you want to have default values, put that in the parameter definition...
[Parameter(Mandatory=$false)]$X=1.00

Reply:
> _If you want to have default values, put that in the parameter definition... [Parameter(Mandatory=$false)]$X=1.00_

I genuinely didn't know you could do that `#everydayisaschoolday` :-)

I will amend the `$X` and `$Y` defaults when I have some free time later in the week/weekend. Thx